packages/flutter_tools/lib/src/reporting/github_template.dart

```diff
@@ -0,0 +1,139 @@
+  ///
+  /// Shorten the URL, if possible.
+  Future<String> toolCrashIssueTemplateGitHubURL(
+      String command,
+      String errorString,
+      String exception,
+      StackTrace stackTrace,
+      String doctorText
+    ) async {
+    final String title = '[tool_crash] $errorString';
+    final String body = '''## Command
+  ```
+  $command
+  ```
+
+  ## Steps to Reproduce
+  1. ...
+  2. ...
+  3. ...
+
+  ## Logs
+  $exception
+  ```
+  ${LineSplitter.split(stackTrace.toString()).take(20).join('\n')}
+  ```
+  ```
+  $doctorText
+  ```
+
+  ## Flutter Application Metadata
+  ${_projectMetadataInformation()}
+  ''';
+
+    final String fullURL = 'https://github.com/flutter/flutter/issues/new?'
+      'title=${Uri.encodeQueryComponent(title)}'
+      '&body=${Uri.encodeQueryComponent(body)}'
+      '&labels=${Uri.encodeQueryComponent('tool,severe: crash')}';
+
+    return await _shortURL(fullURL);
+  }
+
+  /// Provide information about the Flutter project in the working directory, if present.
+  String _projectMetadataInformation() {
+    FlutterProject project;
+    try {
+      project = FlutterProject.current();
+    } on Exception catch (exception) {
+      // pubspec may be malformed.
+      return exception.toString();
+    }
+    try {
+      final FlutterManifest manifest = project?.manifest;
+      if (project == null || manifest == null || manifest.isEmpty) {
+        return 'No pubspec in working directory.';
+      }
+      String description = '';
+      description += '''
+**Version**: ${manifest.appVersion}
+**Material**: ${manifest.usesMaterialDesign}
+**Android X**: ${manifest.usesAndroidX}
+**Module**: ${manifest.isModule}
+**Plugin**: ${manifest.isPlugin}
+**Android package**: ${manifest.androidPackage}
+**iOS bundle identifier**: ${manifest.iosBundleIdentifier}
+''';
+      final File file = project.flutterPluginsFile;
+      if (file.existsSync()) {
+        description += '### Plugins\n';
+        for (String plugin in project.flutterPluginsFile.readAsLinesSync()) {
+          final List<String> pluginParts = plugin.split('=');
+          if (pluginParts.length != 2) {
+            continue;
+          }
+          description += pluginParts.first;
+        }
+      }
+
+      return description;
+    } on Exception catch (exception) {
+      return exception.toString();
+    }
+  }
+
+  /// Shorten GitHub URL with git.io API.
+  ///
+  /// See https://github.blog/2011-11-10-git-io-github-url-shortener.
+  Future<String> _shortURL(String fullURL) async {
+    String url;
+    try {
+      printTrace('Attempting git.io shortener: $fullURL');
+      final List<int> bodyBytes = utf8.encode('url=${Uri.encodeQueryComponent(fullURL)}');
+      final HttpClientRequest request = await _client.postUrl(Uri.parse('https://git.io'));
+      request.headers.set(HttpHeaders.contentLengthHeader, bodyBytes.length.toString());
+      request.add(bodyBytes);
```

Comment:
I didn't realize this code existed until just now... surely this is potentially sending PII to GitHub? This seems like a problem.